src/MatchingResult.php

```diff
@@ -16,10 +18,6 @@ final class MatchingResult implements MiddlewareInterface
     private array $methods = [];
     private ?DispatcherInterface $dispatcher = null;
 
-    private function __construct()
```

Comment:
Why did you remove the private constructor? This class should only be instantiated by factory methods.

Reply:
Thanks. Fixed.